pkg/discovery/default_monitor.go

```diff
@@ -51,14 +51,14 @@ func (monitor *DefaultMesosMonitor) Monitor(target *MonitorTarget) error {
 	arrOfExec, err := monitor.getAgentStats(agent, masterConf)
 	glog.V(3).Infof("Parsed executors %s\n", arrOfExec)
 	if err != nil {
-		glog.Errorf("Error obtaining metrics from the agent %s::%s : %s", agent.IP, agent.Port, err)
-		return fmt.Errorf("Error obtaining metrics from the agent %s::%s", agent.IP, agent.Port)
+		glog.Errorf("Error obtaining metrics from the agent %s::%s : %s", agent.IP, agent.PortNum, err)
+		return fmt.Errorf("Error obtaining metrics from the agent %s::%s", agent.IP, agent.PortNum)
 	}
 
 	err = monitor.parseAgentUsedStats(agent, arrOfExec, target.rawStatsCache)
 	if err != nil {
-		glog.Errorf("Error parsing metrics from the agent %s::%s : %s", agent.IP, agent.Port, err)
-		return fmt.Errorf("Error parsing metrics from the agent %s::%s", agent.IP, agent.Port)
+		glog.Errorf("Error parsing metrics from the agent %s::%s : %s", agent.IP, agent.PortNum, err)
```

Comment:
reduce repeated code by: 

nerr := fmt.Errorf("Error parsing metrics from the agent %s::%s", agent.IP, agent.PortNum)
glog.Errorf("%s : %s", nerr.Error(), err)
return nerr